p2p/p2p-core/src/services.rs

```diff
@@ -86,34 +90,77 @@ pub enum AddressBookRequest<Z: NetworkZone> {
         /// The peers rpc credits per hash
         rpc_credits_per_hash: u32,
     },
+
     /// Tells the address book about a peer list received from a peer.
     IncomingPeerList(Vec<ZoneSpecificPeerListEntryBase<Z::Addr>>),
+
     /// Takes a random white peer from the peer list. If height is specified
     /// then the peer list should retrieve a peer that should have a full
     /// block at that height according to it's pruning seed
     TakeRandomWhitePeer { height: Option<usize> },
+
     /// Takes a random gray peer from the peer list. If height is specified
     /// then the peer list should retrieve a peer that should have a full
     /// block at that height according to it's pruning seed
     TakeRandomGrayPeer { height: Option<usize> },
+
     /// Takes a random peer from the peer list. If height is specified
     /// then the peer list should retrieve a peer that should have a full
     /// block at that height according to it's pruning seed.
     ///
     /// The address book will look in the white peer list first, then the gray
     /// one if no peer is found.
     TakeRandomPeer { height: Option<usize> },
+
     /// Gets the specified number of white peers, or less if we don't have enough.
     GetWhitePeers(usize),
+
+    /// Get the amount of white & grey peers.
+    PeerlistSize,
+
+    /// Get the amount of incoming & outgoing connections.
+    ConnectionCount,
+
+    /// (Un)ban a peer.
+    SetBan(SetBan<Z::Addr>),
+
     /// Checks if the given peer is banned.
-    IsPeerBanned(Z::Addr),
+    GetBan(Z::Addr),
+
+    /// Get the state of all bans.
+    GetBans,
```

Comment:
### Signatures (address book request)
Added to the existing `AddressBookRequest`.

`IsPeerBanned` can be replaced by `GetBan` (see below)